src/idlcxx/src/traits.c

```diff
@@ -207,26 +207,28 @@ emit_traits(
     " };\n"
     "template<> constexpr const unsigned char TopicTraits<%1$s>::type_info_blob[] = {\n";
   const idl_struct_t *_struct = node;
+  const idl_union_t *_union = node;
```

Comment:
It is a bit of a nitpick, but it appears that `_struct` and `_union` serve no other purpose than in defining `selfcontained` and `ext`. Given that `node` can't be both a struct and a union at the same time, having these two defined all the time seems like asking for confusion.

Would you mind rewriting it a little bit, e.g., like:
```C
bool selfcontained;
idl_extensibility_t ext;
if (idl_is_struct(node)) {
  const idl_struct_t *_struct = node;
  selfcontained = sc_struct(_struct);
  ext = _struct->extensibility.value;
} else {
  assert(idl_is_union(node));
  ...
}
```
That would make me sleep a little better at night 😀

Otherwise it LGTM

Reply:
Will rewrite it into a function.